assets/js/Ioda/pages/dashboard/Dashboard.js

```diff
@@ -1,229 +1,227 @@
-      this.props.suggestedSearchResults !== prevProps.suggestedSearchResults
-    ) {
-      this.setState({
-        suggestedSearchResults: this.props.suggestedSearchResults,
-      });
+  useEffect(() => {
+    if (props.suggestedSearchResults) {
+      setSearchResults(props.suggestedSearchResults);
     }
-
-    // After API call for outage summary data completes, pass summary data to map function for data merging
-    if (this.props.summary !== prevProps.summary) {
-      this.setState({ summaryDataRaw: this.props.summary }, () => {
-        this.getMapScores();
-        this.convertValuesForSummaryTable();
-        if (this.state.activeTabType === asn.type) {
-          this.getDataEvents(this.state.activeTabType);
-        }
-
-        if (!this.state.eventEndpointCalled) {
-          this.setState(
-            { eventEndpointCalled: !this.state.eventEndpointCalled },
-            () => {
-              const totalEventCount = this.state.summaryDataRaw.reduce(
-                (acc, item) => acc + item.event_cnt,
-                0
-              );
-              //Get total event count to reference with event data
-              this.setState({ totalEventCount });
-            }
-          );
-        }
-      });
+  }, [props.suggestedSearchResults]);
+
+  useEffect(() => {
+    if(!summary) return;
+    setSummaryDataRaw(summary);
+  }, [summary]);
+
+  useEffect(() => {
+    if(!summaryDataRaw) return;
+    getMapScores();
+    _convertValuesForSummaryTable();
+    if (activeTabType === asn.type) {
+      getDataEvents(activeTabType);
     }
+    if (!eventEndpointCalled) {
+      setEventEndpointCalled(true);
+    }
+  }, [summaryDataRaw, activeTabType, eventEndpointCalled]);
+
+  useEffect(() => {
+    if(!summary) return;
+    if(!eventEndpointCalled) return;
+    if(!summaryDataRaw) return;
+    const totalEventCount = summaryDataRaw.reduce(
+        (acc, item) => acc + item.event_cnt,
+        0
+    );
+    setTotalEventCount(totalEventCount);
+  }, [eventEndpointCalled]);
 
-    // After API call for total outages summary data completes, pass total count to table to populate in UI
-    if (this.props.totalOutages !== prevProps.totalOutages) {
-      this.setState({
-        totalOutages: this.props.totalOutages.length,
-      });
+  useEffect(() => {
+    if (props.totalOutages) {
+      setTotalOutagesState(props.totalOutages);
     }
+  }, [props.totalOutages]);
 
-    // Make API call for data to populate time series stacked horizon view
-    if (this.props.eventSignals !== prevProps.eventSignals) {
-      let newEventData = this.props.eventSignals;
-      this.setState(
-        (prevState) => ({
-          eventDataRaw: [...prevState.eventDataRaw, newEventData],
-        }),
-        () => {
-          this.convertValuesForHtsViz();
-        }
-      );
+  useEffect(() => {
+    if (eventSignals) {
+      setEventDataRaw((prevEventDataRaw) => [...prevEventDataRaw, eventSignals]);
+      convertValuesForHtsViz();
```

Comment:
React state updates are asynchronous, so calling the `convertValuesForHtsViz` is not guaranteed to work the way you expect. This is why it was called in the `this.setState` callback in the previous version. You may need an effect for this.